designs/serialization.md

```diff
@@ -523,3 +523,169 @@ client call to a service. A service could, for example, model its event
 structures and include them in their client. A customer could then use the
 generated `DeserializeableShape`s to deserialize those events into Python types
 when they're received without having to do so manually.
+
+## Codecs
+
+Serializers and deserializers are never truly disconnected - where there's one,
+there's always the other. They need to be tied together in a way that makes
+sense, is portable, and which provides extra utility for common use cases.
+
+One such use case is the serialization and deserialization to and from discrete
+bytes of a common format represented by a media type such as `application/json`.
+These will be represented by the `Codec` interface:
+
+```python
+@runtime_checkable
+class Codec(Protocol):
+
+    @property
+    def media_type(self) -> str:
```

Comment:
FWIW, we removed media_type from codecs in our internal design docs. It's really the protocol that dictates this. e.g., a codec might be for JSON, but the protocol actually requires an AWS JSON media type.